test/kitchen/test/integration/dd-system-probe-test/rspec/dd-system-probe-test_spec.rb

```diff
@@ -1,15 +1,24 @@
 require 'spec_helper'
+require 'open3'
 
 print `cat /etc/os-release`
 print `uname -a`
 
+GOLANG_TEST_FAILURE = /FAIL:/
+
 Dir.glob('/tmp/system-probe-tests/**/testsuite').each do |f|
   describe "system-probe tests for #{f}" do
     it 'succesfully runs' do
       Dir.chdir(File.dirname(f)) do
-        `sudo #{f} -test.v 1>&2`
-        retval = $?
-        expect(retval).to eq(0)
+        Open3.popen2e("sudo", f, "-test.v") do |_, output, _|
+          test_failures = []
+          output.each_line do |line|
+            puts "#{line}"
+            test_failures << line.strip if line =~ GOLANG_TEST_FAILURE
+          end
+
+          expect(test_failures).to be_empty, test_failures.join("\n")
```

Comment:
Can we keep the return value check as well? I imagine there are scenarios where some the test binary fails immediately that wouldn't have `FAIL:` in the output?

Reply:
Can do